Command/GenerateTokenCommand.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Command;
+
+use Lexik\Bundle\JWTAuthenticationBundle\Services\JWTTokenManagerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Security\Core\User\User;
+
+/**
+ * GenerateTokenCommand.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class GenerateTokenCommand extends Command
+{
+    protected static $defaultName = 'lexik:jwt:generate-token';
+
+    private $tokenManager;
+
+    public function __construct(JWTTokenManagerInterface $tokenManager)
+    {
+        parent::__construct();
+
+        $this->tokenManager = $tokenManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName(static::$defaultName)
+            ->setDescription('Generate a JWT token')
+            ->addArgument('username', InputArgument::REQUIRED)
+            ->addOption('user-class', 'c', InputOption::VALUE_REQUIRED, 'Class of the User to use.', User::class)
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $userClass = $input->getOption('user-class');
+        $token     = $this->tokenManager->create(new $userClass($input->getArgument('username'), null));
```

Comment:
my User entity does not accept any arguments in its constructor, so this will most likely fail with my specific `user-class`


FYI, when I make the same command in my projects, this is how I retrieve the token:
```php
        $user = $this->userProvider->loadUserByUsername($input->getArgument('username'));
        $token = $this->tokenManager->create($user);
```

where `userProvider` is a `UserProviderInterface` which I map to my user provider service like this:

```yaml
services:
    _defaults:
        autowire: true
    # ...
    Symfony\Component\Security\Core\User\UserProviderInterface: '@security.user.provider.concrete.entity_provider'

# note: "entity_provider" is the name of my provider in "security.yaml"
```

So, maybe this would be safer to have a `user-provider-class` option instead of `user-class` (or even configuring this to the `lexik_jwt_authentication.yaml` file)
, since we can check that the given class implements the `UserProviderInterface`

Reply:
I agree that the current solution is not ideal, but I'm not fond of `user-provider-class` either as it implies to have a service named accordingly, whereas you could have several providers based on the same class (e.g. an `entity` provider for two different entities).

What about giving all configured user providers to the command (as a service locator, indexed by their short name) and change the option name to `user-provider`? 
In case no `--user-provider` is passed, if there is only one configured provider then use it, fail otherwise  (saying that the option must be passed since multiple providers exist).
Providers can be retrieved from the context listener definition, see https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php#L180.
I can help if you like it